client/allocrunnerv2/taskrunner/logmon_hook.go

```diff
@@ -0,0 +1,101 @@
+package taskrunner
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"runtime"
+
+	hclog "github.com/hashicorp/go-hclog"
+	plugin "github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/nomad/client/allocdir"
+	"github.com/hashicorp/nomad/client/allocrunnerv2/interfaces"
+	"github.com/hashicorp/nomad/client/logmon"
+	"github.com/hashicorp/nomad/helper/uuid"
+)
+
+// logmonHook launches logmon and manages task logging
+type logmonHook struct {
+	// logmon is the handle to the log monitor process for the task.
+	logmon             logmon.LogMon
+	logmonPluginClient *plugin.Client
+
+	stdoutFifo string
+	stderrFifo string
+
+	taskDir *allocdir.TaskDir
+
+	logger hclog.Logger
+}
+
+func newLogMonHook(tr *TaskRunner, logger hclog.Logger) *logmonHook {
+	hook := &logmonHook{
+		taskDir: tr.taskDir,
+		logger:  logger,
+	}
+
+	tr.taskLoggingFifoGetter = hook.getFifos
```

Comment:
I am not a huge fan of this. Can we instead create a config struct for the hook and the config has a method to SetFifos and you can move the OS selection logic there and that function returns the paths as well and the TR can store it.